src/ServicePulse.Host/app/js/views/pending_retries/controller.js

```diff
@@ -0,0 +1,375 @@
+﻿/// <reference path="service.js" />
+;
+(function(window, angular, undefined) {
+    "use strict";
+
+    function controller(
+        $scope,
+        $timeout,
+        $location,
+        $moment,
+        $filter,
+        scConfig,
+        toastService,
+        sharedDataService,
+        notifyService,
+        serviceControlService,
+        endpointsService, 
+        redirectService,
+        pendingRetryService) {
+
+        var vm = this;
+
+        var notifier = notifyService();
+        
+        vm.sortButtonText = '';
+        vm.sortDirection = 'asc';
+        vm.allMessagesLoaded = false;
+        vm.loadingData = false;
+
+        vm.timeGroup = {
+            amount: undefined,
+            unit: undefined,
+            buttonText: 'All Pending Retries',
+            selected: function () {
+                return $moment.duration(vm.timeGroup.amount, vm.timeGroup.unit);;
+            }
+        };
+
+        notifier.subscribe($scope, function (event, data) {
```

Comment:
It might be better to rename `data` to something meaningful here. Maybe `updatedTotalMessages`? Or something that reflects what it is. Makes the code easier to follow.